metrics/internal/cache/rbd-mirror.go

```diff
@@ -177,7 +177,7 @@ func initCeph(kubeclient clientset.Interface, allowedNamespaces []string) (cephM
 	}
 	id := "admin"
 
-	configmap, err := kubeclient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), "rook-ceph-csi-config", v1.GetOptions{})
+	configmap, err := kubeclient.CoreV1().ConfigMaps(namespace).Get(context.TODO(), "rook-ceph-mon-endpoints", v1.GetOptions{})
```

Comment:
@Madhu-1 In case of external mode clusters configmap `rook-ceph-csi-config` in external cluster namespace is empty. But, configmap `rook-ceph-mon-endpoints` in same namespace is not empty and has the same details.
However, configmap `rook-ceph-csi-config` in operator namespace has 2 csi config details, 1 for internal mode and 1 for external mode. Is this expected?

For now I have moved to configmap `rook-ceph-mon-endpoints` as it's behavior is most consistent. But, maybe it's an issue that needs to be fixed for rook/ceph-csi.

Reply:
@umangachapagain do we have a clusters or some sample data to compare both, i assume `rook-ceph-csi-config` should also contains same details in case of both internal/external mode as well. https://github.com/rook/rook/blob/df04e5cce53f7b4c34cc060877b09cae2a5e4ffc/pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml#L247-L250 is the configmap mounted to csi pod which contains the mapping details

Reply:
CSI is deployed in operator namespace. So, even in case of internal+external, it will use the configmap from operator namespace for both clusters. Since that ConfigMap has needed details, nothing breaks. I am assuming CSI code can handle cases where key `csi-cluster-config-json` has multiple distinct values.

Reply:
i think using `rook-ceph-mon-endpoints` might not work if we have different subvolumegroups and radosnamespaces in the cluster, it will contains only the default entry am also not sure who is using that entry from monitor configmap. At just a glance that a dummy entry is monitor configmap which no one uses. you might need to switch back to `rook-ceph-csi-config` for csi operations.

Reply:
I am using the configmap only to get the clusterID and MON IPs for each cluster. What's the most reliable way to do that? Something that will work consistently between internal and external modes.
Maybe we need to fix the `rook-ceph-csi-config` behavior.

Reply:
@travisn Do you have any suggestions on this?

Reply:
>I am using the configmap only to get the clusterID and MON IPs for each cluster. What's the most reliable way to do that? Something that will work consistently between internal and external modes.
Maybe we need to fix the rook-ceph-csi-config behavior.

It should be same there wont be any problem because cephcsi uses this configmap in case of both internal/external. can we get some Yaml from both cases to check what exact data you are referring to?

@umangachapagain this doesnt work for the cases where we will have subvolumegroup or the radosnamespace, AFAIK `rook-ceph-mon-endpoints` will not get updated for the same (this might be bug in Rook)

Reply:
I have updated code to use `rook-ceph-csi-config` from the operator namespace which is the updated one and also used by Ceph-CSI.